src/__tests__/output-elm.spec.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  ComponentCompilerMeta,
+  ComponentCompilerPropertyType,
+  Encapsulation,
+} from '@stencil/core/internal/stencil-private';
+import { generateProxyElmModule } from '../output-elm';
+
+describe('generateProxyElmModule', () => {
```

Comment:
Nitpicking - but somewhere in this block would be a good place to put a `toMatchSnapshot` assertion. This is the kind of scenario where that function is most useful.

Reply:
Disagree, I think. I don't like snapshot tests where the test does not describe the behaviour it is intended to test. These make it very hard for future maintainers to know if a failing test is a problem or not, and whether it's safe to generate a new snapshot.

I much prefer my approach here that tests the parts of the output that I care about, and makes tests that describe the intended behaviour of the system.

Reply:
I definitely agree that snapshot tests can be noise that isn't always the best indicator of issues. Unintended side-effects would be the kinds of issues I'd expect snapshots to catch - e.g. maybe a change adds another attribute to `Props` but it's not reflected in view. 

Personally I would throw one in here, but I understand why you would not (hence the /nit :D)